services/wheelmap/wheelmap.service.js

```diff
@@ -1,59 +1,80 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const serverSecrets = require('../../lib/server-secrets')
+
+const wheelmapSchema = Joi.object({
+  node: Joi.object({
+    wheelchair: Joi.string().required(),
+  }).required(),
+}).required()
+
+module.exports = class Wheelmap extends BaseJsonService {
+  async fetch({ nodeId }) {
+    let options
+    if (serverSecrets.wheelmap_token) {
+      options = {
+        qs: {
+          api_key: `${serverSecrets.wheelmap_token}`,
+        },
+      }
+    }
+
+    return this._requestJson({
+      schema: wheelmapSchema,
+      url: `http://wheelmap.org/api/nodes/${nodeId}`,
+      options,
+      errorMessages: {
+        401: 'invalid token',
+        404: 'node not found',
+      },
+    })
+  }
+
+  static render({ accessibility }) {
+    let color
+    if (accessibility === 'yes') {
+      color = 'brightgreen'
+    } else if (accessibility === 'limited') {
+      color = 'yellow'
+    } else if (accessibility === 'no') {
+      color = 'red'
+    }
+    return { message: accessibility, color }
+  }
+
+  async handle({ nodeId }) {
+    const json = await this.fetch({ nodeId })
+    const accessibility = json.node.wheelchair
+    return this.constructor.render({ accessibility })
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'accessibility' }
+  }
 
-module.exports = class Wheelmap extends LegacyService {
   static get category() {
     return 'other'
   }
 
   static get route() {
     return {
       base: 'wheelmap/a',
+      format: '(-?[0-9]+)',
```

Comment:
Could this use a `pattern`? I'm trying to convert `format`s to `pattern`s wherever we can.